docs/installation/prepare/microcontrollers/_index.md

```diff
@@ -1,387 +1,353 @@
+
+Copy the following JSON template and paste it into your configuration:
+
+{{< tabs name="Configure an esp32 Board" >}}
+{{% tab name="JSON Template"%}}
+
+```json
+{
+  "attributes": {
+    "pins": [
+      <int>
+    ],
+    "analogs": [
+      {
+        "pin": "<number>", 
+        "name": "<your-analog-name>"
+      }
+    ]
+  },
+  "depends_on": [],
+  "model": "esp32",
+  "name": "<your-board-name>",
+  "type": "board"
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "attributes": {
+    "pins": [
+      15
+    ],
+    "analogs": [
+      {
+        "pin": "34",
+        "name": "sensor"
+      }
+    ]
+  },
+  "depends_on": [],
+  "model": "esp32",
+  "name": "board",
+  "type": "board"
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Edit and fill in the attributes as applicable.
+Click **Save config**.
+The following attributes are available for `esp32` boards:
+
+| Name | Type | Inclusion | Description |
+| ---- | ---- | --------- | ----------- |
+| `pins` | object | Required | The {{< glossary_tooltip term_id="pin-number" text="pin number" >}} of any GPIO pins you wish to use as input/output with the [`GPIOPin` API](/program/apis/#gpio-pins). |
+| `analogs` | object | Optional | Attributes of any pins that can be used as analog-to-digital converter (ADC) inputs. See [configuration info](/components/board/#analogs). |
+<!-- | `webhook` | string | Optional | A webhook's URL. See [GitHub](viam-labs/webhook-template) for an example. |
+| `webhook-secret` | string | Optional | The robot secret for a configured `webhook`. See [GitHub](viam-labs/webhook-template) for an example. | -->
+
+### Generate a New Project from the Micro-RDK Template
+
+Use [the Micro-RDK template](https://github.com/viamrobotics/micro-rdk-robot-template.git) to create a new Micro-RDK project to upload to your ESP32 by running:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo generate --git https://github.com/viamrobotics/micro-rdk-robot-template.git
+```
+
+You will be prompted to paste your robot's JSON configuration into the terminal.
+To obtain this:
+
+* Navigate to [your new robot's](/installation/prepare/microcontrollers/#create-a-new-robot) page on [the Viam app](https://app.viam.com).
+* Click the **Copy viam-server config** button on the right side of the **Setup** tab.
+The micro-RDK needs this JSON file, which contains your robot part secret key and cloud app address, to connect to [the Viam app](https://app.viam.com).
+* Paste the `viam-server` config into your terminal when prompted.
+
+{{% snippet "secret-share.md" %}}
+
+### Connect to your ESP32
+
+Now, upload the project to connect to your ESP32 through [the Viam app](https://app.viam.com):
+
+{{< tabs >}}
+{{% tab name="Use Canon" %}}
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+canon bash -lc "make build-esp32-bin"
```

Comment:
could be me...but i can't past this step on the macos:
```
14:26 $ canon bash -lc "make build-esp32-bin"
Started new container: canon-viam-micro-rdk-acc408ce
Error: http: invalid Host header
```

Reply:
know bug see https://viaminc.slack.com/archives/C04TJK43WA0/p1689263776131489 fix exist should be merged in the short term so unsure we should mentioned it 
run `docker run -it --rm --user testbot -v "${PWD}:/host" [ghcr.io/viamrobotics/micro-rdk-dev-env:amr64](http://ghcr.io/viamrobotics/micro-rdk-dev-env:amr64) bash -lc "make build-esp32-bin"` in the mean time

Reply:
replace arm64 by amd64 if on x86

Reply:
perfect. thank you.
yeah, not sure if we should mention it, but it's also very needed for anyone trying today.

Reply:
Docker JUST (an hour ago) released a new version to fix the bug, and I'll be updating canon to use it soon. Hopefully should be no need to mention it in docs.